generic_server.cc

```diff
@@ -363,7 +380,13 @@ future<> server::do_accepts(int which, bool keepalive, socket_address server_add
             fd.set_nodelay(true);
             fd.set_keepalive(keepalive);
             auto conn = make_connection(server_addr, std::move(fd), std::move(addr),
-                    _conns_cpu_concurrency_semaphore, {});
+                    _conns_cpu_concurrency_semaphore, std::move(units));
+            if (shed) {
+                _logger.debug("too many in-flight connection attempts: {}, connection dropped",
+                        _conns_cpu_concurrency_semaphore.waiters());
+                conn->on_connection_close();
```

Comment:
I guess this answers "what if the connection breaks". I hope it's fully accounted or we can end up with a hung server. Maybe move to the destructor to increase robustness?